.github/workflows/ci.yml

```diff
@@ -0,0 +1,192 @@
+      # MacOS
+      - name: Install dependencies for MacOS
+        if: ${{ contains(matrix.os, 'macos') }}
+        run: brew install fftw
+
+      - name: Fix omp headers not linked on MacOS
+        if: ${{ contains(matrix.os, 'macos') }}
+        run: |
+          brew unlink libomp
+          brew link --force libomp
+          export DOMP_ROOT_MAC=-DOMP_ROOT=$(brew --prefix libomp)
+          echo "DOMP_ROOT_MAC=${DOMP_ROOT_MAC}" >> $GITHUB_ENV
+
+      # -------------------------------------------------------------------------------
+      - name: Create build environment
+        run: |
+          cmake -E make_directory ${{ runner.workspace }}/build
+
+      - name: Compile with Werror on Ubuntu and MacOS (not Windows)
+        if: ${{ contains(matrix.os, 'macos') }} || ${{ contains(matrix.os, 'ubuntu')}}
+        run: echo "CXXFLAGS=-Werror" >> $GITHUB_ENV
+
+      - name: Configure TDMS
+        shell: bash
+        working-directory: ${{ runner.workspace }}/build
+        run: |
+          cmake ${GITHUB_WORKSPACE}/tdms -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+           -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+           -DBUILD_TESTING=${{ matrix.build_testing }} \
+           ${DOMP_ROOT_MAC}
+        # This 👆 env variable only exists on MacOS builds to fix the OMP
+        # homebrew/PATH problem.
+
+      - name: Build TDMS
+        working-directory: ${{ runner.workspace }}/build
+        shell: bash
+        run: |
+          cmake --build . --config ${{ matrix.build_type }}
+
+      - name: Run TDMS unit tests
+        if: matrix.build_testing == 'ON'
+        working-directory: ${{ runner.workspace }}/build
+        shell: bash
+        run: |
+          export OMP_NUM_THREADS=1
+          ctest -C ${{ matrix.build_type }} --output-on-failure --extra-verbose
+
+      - name: Tar the build result to maintain permissions
+        # https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files
+        # https://github.com/actions/upload-artifact/issues/38
+        if: ${{ !contains(matrix.os, 'windows') }}
+        working-directory: ${{ runner.workspace }}/build
+        run: tar -cvf tdms_build.tar tdms
+
+      - name: Upload build results
+        if: matrix.build_testing == 'OFF' && matrix.build_type == 'Debug'
+        # If we use the build with testing ON, we'd also need to copy over (and
+        # install) the libtdms.so so use the OFF build for simplicity.
+        uses: actions/upload-artifact@v3
+        with:
+          name: tdms_build_${{ matrix.os }}
+          path: ${{ runner.workspace }}/build/tdms_build.tar
+          retention-days: 1
+
+  # Only run the slow system tests on Ubuntu.
+  # Cache the input data to save network usage.
+  system-tests:
+    name: System tests ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest] #, macos-latest]
```

Comment:
I think `fail-fast` being false for the `build` matrix makes sense: if it's an OS-specific problem we get notified of it, and the `system test`s don't start regardless.

That is, I'd rather know how many out of the 6 `builds` worked, rather than only being able to see 1 failed, so the rest stopped.